service/history/handler.go

```diff
@@ -1436,9 +1433,9 @@ func (h *Handler) convertError(err error) error {
 			return serviceerrors.NewShardOwnershipLost(h.GetHostInfo().GetAddress(), info.GetAddress())
 		}
 		return serviceerrors.NewShardOwnershipLost(h.GetHostInfo().GetAddress(), "<unknown>")
-	case *persistence.WorkflowExecutionAlreadyStartedError:
-		err := err.(*persistence.WorkflowExecutionAlreadyStartedError)
-		return serviceerror.NewWorkflowExecutionAlreadyStarted(err.Msg, err.StartRequestID, err.RunID)
+	case *persistence.WorkflowConditionFailedError:
+		err := err.(*persistence.WorkflowConditionFailedError)
+		return serviceerror.NewInternal(err.Msg)
```

Comment:
If we assign in the switch statement (e.g. `switch err := err.(type)`) then we don't have to perform the cast in every `case`

Reply:
good point, this change is probably due to s/WorkflowExecutionAlreadyStarted/WorkflowConditionFailedError/g